src/main/java/com/getbase/sync/Sync.java

```diff
@@ -52,12 +52,18 @@ public boolean fetch(BiPredicate<Meta, Map<String, Object>> predicate) {
         return fetchInternal(predicate);
     }
```

Comment:
Instead of exposing `SyncProcess` we can add another ctor to `Sync` which will take `timeout`. The existing ctor will assume timeout is `Integer.MAX_VALUE`.